sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosDataSource.scala

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark
+
+import java.util
+
+import org.apache.spark.sql.connector.catalog.{Table, TableProvider}
+import org.apache.spark.sql.connector.expressions.Transform
+import org.apache.spark.sql.sources.DataSourceRegister
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
+
+class CosmosDataSource extends DataSourceRegister with TableProvider with CosmosLoggingTrait {
+  logInfo(s"Instantiated ${this.getClass.getSimpleName}")
+
+  override def inferSchema(caseInsensitiveStringMap: CaseInsensitiveStringMap): StructType = {
+    getTable(null,
+      Array.empty[Transform],
+      caseInsensitiveStringMap.asCaseSensitiveMap()).schema()
+  }
+
+  override def shortName(): String = "cosmos.write"
```

Comment:
In my mental model I though about something like

cosmos.items (which would implement the inetrfaces for read and write(batch and point))

cosmos.changefeed for changefeed - just read interfaces

Reply:
"cosmos.items" for batch read/write looks better to me.
regarding "cosmos.changefeed", that will be used mainly for streaming scenario right? should we add streaming suffix?

Reply:
done

Reply:
Changefeed would be used for both streaming and batch (mostly streaming i guess but we also have plenty of customers using batch in regular intervals)
Streaming and batch are just different notions on the read write capabilities so I would not add any prefixes there
But between Items and Changefeed there are significant differences - write on CF isn't possible and even read is very different because for example predicate push down isn't possible for CF etc.